includes/data-stores/class-wc-data-store-wp.php

```diff
@@ -243,39 +243,181 @@ protected function get_wp_query_args( $query_vars ) {
-		if ( ! $query_var ) {
-			return '';
-		}
+	protected function parse_date_for_wp_query( $query_var, $key, $wp_query_args = array() ) {
+		$query_parse_regex = '/([^.<>]*)(>=|<=|>|<|\.\.\.)([^.<>]+)/';
+		$valid_operators = array( '>', '>=', '=', '<=', '<', '...' );
+
+		// YYYY-MM-DD queries have 'day' precision. Timestamp/WC_DateTime queries have 'second' precision.
+		$precision = 'second';
+
+		$start_date = null;
+		$operator = '=';
+		$end_date = null;
 
+		// Specific time query with a WC_DateTime.
 		if ( is_a( $query_var, 'WC_DateTime' ) ) {
-			$datetime = $query_var;
+			$end_date = $query_var;
+
+		// Specific time query with a timestamp.
 		} elseif ( is_numeric( $query_var ) ) {
-			// Timestamps are handled as UTC timestamps in all cases.
-			$datetime = new WC_DateTime( "@{$value}", new DateTimeZone( 'UTC' ) );
+			$end_date = new WC_DateTime( "@{$query_var}", new DateTimeZone( 'UTC' ) );
+
+		// Query with operators and possible range of dates.
+		} elseif ( preg_match( $query_parse_regex, $query_var, $sections ) ) {
+			if ( ! empty( $sections[1] ) ) {
+				$start_date = is_numeric( $sections[1] ) ? new WC_DateTime( "@{$sections[1]}", new DateTimeZone( 'UTC' ) ) : wc_string_to_datetime( $sections[1] );
+			}
+
+			$operator = in_array( $sections[2], $valid_operators ) ? $sections[2] : '';
+			$end_date = is_numeric( $sections[3] ) ? new WC_DateTime( "@{$sections[3]}", new DateTimeZone( 'UTC' ) ) : wc_string_to_datetime( $sections[3] );
+
+			if ( ! is_numeric( $sections[1] ) && ! is_numeric( $sections[3] ) ) {
+				$precision = 'day';
+			}
+
+		// Specific time query with a string.
 		} else {
-			// Strings are defined in local WP timezone. Convert to UTC.
-			if ( 1 === preg_match( '/^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})(Z|((-|\+)\d{2}:\d{2}))$/', $query_var, $date_bits ) ) {
-				$offset    = ! empty( $date_bits[7] ) ? iso8601_timezone_to_offset( $date_bits[7] ) : wc_timezone_offset();
-				$timestamp = gmmktime( $date_bits[4], $date_bits[5], $date_bits[6], $date_bits[2], $date_bits[3], $date_bits[1] ) - $offset;
+			$end_date = wc_string_to_datetime( $query_var );
+			$precision = 'day';
+		}
+
+		// Check for valid inputs.
+		if ( ! $operator || ! $end_date || ( '...' === $operator && ! $start_date ) ) {
+			return $wp_query_args;
+		}
+
+		// Build date query for 'post_date' or 'post_modified' keys.
+		if ( 'post_date' == $key || 'post_modified' == $key ) {
+			if ( ! isset( $wp_query_args['date_query'] ) ) {
+				$wp_query_args['date_query'] = array();
+			}
+
+			$query_arg = array(
+				'column' => $key . '_gmt',
+				'inclusive' => '>' !== $operator && '<' !== $operator,
+			);
+
+			if ( '>' == $operator || '>=' == $operator ) {
+				$query_arg['after'] = array(
+					'year' => $end_date->date( 'Y' ),
+					'month' => $end_date->date( 'n' ),
+					'day' => $end_date->date( 'j' ),
+				);
+				if ( 'second' === $precision ) {
+					$query_arg['after']['minute'] = $end_date->date( 'i' );
```

Comment:
the logic around here is a little hard to read because of the use of end_date/start_date variable names being used for before/after respectively - not sure if it can be improved.

Could just have an array of dates. If 1 date we know it's a comparison/before/after type.

if it's an array we know it's a range. This might reduce logic required.